### DISCUSSION TITLE
IATI Usability Scores

### DISCUSSION THREAD

**Discussion by bill-anderson** [devinit/ddw-analyst-ui#385] [Ideas]
The current scores (for commitments and spend) in the IATI Publishers Table are:
0 (don't use)
1 (review comment before using)
2 (go ahead)

We have seen this from the near-real-time perspective and it doesn't cover publishers whose data is unusable in the current year, but is usable when doing annual aggregations up to the previous year. I therefore suggest we add another score.
0 - don't use
1 - don't use current year, but acceptable for annual aggregates of previous years
2 - current year data is usable with some review and manipulation
3 - usable in near-real-time analysis

**Comment by dean-breed** [comment]
Thanks @bill-anderson

**Comment by bill-anderson** [comment]
Meant to add that the assumption is that to score 2 you must meet 1 as well

**Comment by dean-breed** [comment]
Yes, absolutely and that makes sense. What is the best way to tackle this Bill?

Perhaps if you take a look and score using this system, I could create a set of rules that applies the scoring system automatically. It would obviously need review from yourself to check it is working as we'd like. I assume the long term goal is to have an automatic scoring system for all donors?

**Comment by bill-anderson** [comment]
1. Change all 1's and 2's to 2's and 3's
2. Use the DAC update process to see who can be added with score 1
3. Write and manually test pseudocode for rules
4. Write the automated process

**Comment by dean-breed** [comment]
I'll do 1 today and then start the rest in January. I'll edit the publishers list and add it to issue #365

**Comment by dean-breed** [comment]
Point 1 done. We will need to consider that it is likely some publishers will move from 2 to 1 and 0 to 1, depending on the reasons we had them in 0 or 2 to begin with.
